app/_ui/auth/verification.tsx

```diff
@@ -62,76 +79,103 @@ function EmailSentComponent(props: EmailSentComponentProps) {
               error={(formik.errors.code != undefined)}
               />
       <Box display={'flex'} flexDirection={'row'} justifyContent={'space-between'} width={'100%'}>
-        <Button color="primary" variant="contained" onClick={() => {}}>
+        <Button color="primary" variant="contained" onClick={() => {router.back()}}>
           Back
         </Button>
-        <Button color="primary" variant="contained" onClick={() => {}}>
+        <Button color="primary" variant="contained" onClick={resendCode} disabled={formik.isSubmitting}>
           Resend Code
         </Button>
         <Button color="primary" variant="contained" type="submit" disabled={formik.isSubmitting}>
           Submit
         </Button>
-
       </Box>
     </Box>
     </form>
   )
 }
 
+interface PassedChallengeComponentProps {}
+
+function PassedChallengeComponent(props: PassedChallengeComponentProps): React.ReactElement {
+  return (
+    <Box display={'flex'} flexDirection={'column'} gap={4}>
+      <Typography>Successfully verified account</Typography>
+      <Button href={'portal'}>Continue</Button>
+    </Box>
+    )
+}
+
 interface VerificationComponentProps {
   flow?: IOryVerificationFlow
 }
 
-export function VerificationComponent(props: VerificationComponentProps): React.ReactElement {
-  const [flow, setFlow] = useState<IOryVerificationFlow | undefined>(props.flow)
-  const [content, setContent] = useState<React.ReactElement>()
 
+export function VerificationComponent(props: VerificationComponentProps): React.ReactElement {
   const searchParams = useSearchParams()
+  const pathname = usePathname()
   const flowId = searchParams.get('flow')
+  const [flow, setFlow] = useState<IOryVerificationFlow | undefined>(props.flow)
+  
+  const [content, setContent] = useState<React.ReactElement>(<CircularProgress style={{alignSelf: 'center'}}/>)
+  const router = useRouter()
+  const studyContext = useContext(ProtocolContext)
 
   useEffect(() => {
-    if (flow == undefined) {
-      if (flowId == null) {
-        fetch(withBasePath('/api/ory/verification/browser')).then(
-          (response) => {
-            if (response.ok) {
-              response.json().then(
-                (data) => {
-                  setFlow(data as IOryVerificationFlow)
-                }
-              )
-            }
-          }
-        )
-      } else {
-        fetch(withBasePath(`/api/ory/verification/flows?flow=${flowId}`)).then(
-          (response) => {
-            if (response.ok) {
-              response.json().then(
-                (data) => {
-                  setFlow(data as IOryVerificationFlow)
-                }
-              )
-            }
-          }
-        )
-      }
-    }
+    // if (flow == undefined) {
```

Comment:
Do we not need this anymore?

Reply:
The flow is fetched or created on the server-side page and then passed as a param to this client-side component. It saves a bit of loading time on the client-side while it waits for a response. It might be worth adding in client-side flow support as well, but I can't really think of when it would be useful.

Reply:
Ah okay I see. 👍